File: app/src/main/java/com/lifedawn/bestweather/commons/classes/MainThreadWorker.java
package com.lifedawn.bestweather.commons.classes;

import android.os.Handler;
import android.os.Looper;

import androidx.core.os.HandlerCompat;

public class MainThreadWorker {
	private final static Handler handler = new Handler(Looper.getMainLooper());

	public static void runOnUiThread(Runnable action) {
		handler.post(action);
	}

}

File: app/src/main/java/com/lifedawn/bestweather/commons/classes/MainThreadWorker.kt
package com.lifedawn.bestweather.commons.classes

import android.os.Handler
import android.os.Looper

object MainThreadWorker {
    private val handler = Handler(Looper.getMainLooper())
    @JvmStatic
    fun runOnUiThread(action: Runnable) {
        if (Thread.currentThread() === Looper.getMainLooper().thread) action.run() else handler.post(action)
    }
}